### PR TITLE
Use attached process data

### PR DESCRIPTION
**Pull request by skvl**
This should make _drakvuf_ more accurate.

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/libdrakvuf/vmi.c [hunk 1]
```diff
@@ -446,6 +459,16 @@ event_response_t pre_mem_cb(vmi_instance_t vmi, vmi_event_t* event)
         pass->proc_data.pid       = proc_data.pid;
         pass->proc_data.ppid      = proc_data.ppid;
         pass->proc_data.userid    = proc_data.userid;
+        pass->proc_data.tid       = proc_data.tid;
+        pass->attached_proc_data.base_addr = trap_info.attached_proc_data.base_addr;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
```
Comment (tklengyel) [inline src/libdrakvuf/vmi.c:465]: Could we  get this working without having to disable the warning?

Comment (skvl) [reply src/libdrakvuf/vmi.c]: Yes I can.

Without this I get the error:
```
vmi.c:327:27: error: passing argument 1 of ‘g_free’ discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
     g_free(pass->proc_data.name);
```

But it I remove `const` qualifier from `struct process_data` field `name` (see d9fa5e2) than everything is OK.

Comment (tklengyel) [reply src/libdrakvuf/vmi.c]: Have a local non-const version of variable, you can pass that to g_free.

Comment (skvl) [reply src/libdrakvuf/vmi.c]: @tklengyel I have fixed this in d9fa5e2. You don't like this commit?

Comment (tklengyel) [reply src/libdrakvuf/vmi.c]: Sorry I don't, the proc_data char* field needs to remain const.

File: src/libdrakvuf/libdrakvuf.h [hunk 2]
```diff
@@ -170,7 +170,7 @@ typedef enum memaccess_type
 
 typedef struct process_data
 {
-    const char* name;   /* Process name */
```
Comment (tklengyel) [inline src/libdrakvuf/libdrakvuf.h:173]: This must remain const so a plugin don't accidently free it.

File: src/libdrakvuf/vmi.c (hunk 1 above)
Comment (skvl) [reply src/libdrakvuf/vmi.c]: That's all right. I will fix it.

**Comment by skvl** [comment]
@tklengyel it should be done now (see 7145c3de103c33a33efc6131f509b38aec1ad69d)

**Comment by skvl** [comment]
@tklengyel it should be done now (see 7145c3de103c33a33efc6131f509b38aec1ad69d)

File: src/libdrakvuf/libdrakvuf.h [hunk 3]
```diff
@@ -170,7 +170,7 @@ typedef enum memaccess_type
...
+    char* name;         /* Process name */
```
Comment (tklengyel) [inline src/libdrakvuf/libdrakvuf.h:173]: This must remain `const` here.

**Comment by skvl** [comment]
@tklengyel it should be done now (see 7145c3de103c33a33efc6131f509b38aec1ad69d)

File: src/libdrakvuf/libdrakvuf.h (hunk 3 above)
Comment (tklengyel) [reply src/libdrakvuf/libdrakvuf.h:173]: Why are you marking this resolved when the field is still not `const`?

Comment (tklengyel) [reply src/libdrakvuf/libdrakvuf.h:173]: What I mean for "introduce a non-const" version for the process name is to have it local to the event callbacks in `libdrakvuf/vmi.c`, ie a variable on the stack there. This field should never be exposed to the plugin side as non-const.

Comment (skvl) [reply src/libdrakvuf/libdrakvuf.h]: @tklengyel I'm very sorry but I have forgotten to stage file (at the end of the working day :) )

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

File: src/libdrakvuf/libdrakvuf.h (hunk 3 above)
Comment (tklengyel) [reply src/libdrakvuf/libdrakvuf.h]: OK, no problem :)

**Pull request merged**